### ISSUE TITLE
MD046 vs indented code blocks

### ISSUE THREAD

**Issue by davidkazuhiro** [markdownlint/markdownlint#217]
How can I put a code block in a list and also comply with MD046?

For example

````
You can break the following command into two steps

```
foo && bar
```

as follows:

1. Do foo

    ```
    foo
    ```

1. Do bar

    ```
    bar
    ```
````

violates MD046

**Comment by jaymzh** [comment]
OK the attached PR should fix this and has tests around it.

**Comment by jaymzh** [comment]
Fixed in 0.9.0.

**Comment by ghost** [comment]
I am still able to reproduce this problem on 0.11.0 using [admonitions](https://python-markdown.github.io/extensions/admonition/), for example:

```
!!! note
    This is an admonition with some code:

    ```bash
    echo "code here violates MD046"
    ```

    More text inside the admonition
```
